lib/puma/plugin/statsd.rb

```diff
@@ -8,30 +8,23 @@
 
 Puma::Plugin.create do
 
-  class Statsd
-    ENV_NAME = "STATSD_HOST"
-    STATSD_TYPES = { count: 'c', gauge: 'g' }
-
-    attr_reader :host, :port
+  # uses statsd-ruby client
+  class PumaStatsd
+    attr_reader :host, :client
 
     def initialize
-      @host = ENV.fetch(ENV_NAME, nil)
-      @port = ENV.fetch("STATSD_PORT", 8125)
+      @host = ENV.fetch('APP_STATSD_HOST', '127.0.0.1')
+      port = ENV.fetch('APP_STATSD_PORT', 9125)
+      statsd_config = [@host, @port.to_i]
+      @client = Statsd.new(*statsd_config).tap do |client|
+        # A small hack to add tags. Bypassing the postfix formatting enforced by the statsd client.
+        client.instance_variable_set(:@postfix, ",env=#{App.namespace}")
```

Comment:
`App.namespace` won't exist in this context. If we want it be specific to our application it would be `ENV.fetch('APP_KUBE_NAMESPACE', 'na')`

IMO should avoid putting application specific code here, and I wouldn't want to be loading the rails app anyways if it where in our repo. Is there way we can make those tags configurable through an environment variable and do the same with the rails app? (I'll wait happily wait for someone to say "Eww, it's not DRY" :stuck_out_tongue:)

Maybe something like the following?
In our rails app:
```ruby
tags = ENV.fetch('APP_STATSD_TAGS', "env=#{App.namespace}")
...
client.instance_variable_set(:@postfix, ",#{tags}")
```

Here:
```ruby
tags = ENV.fetch('APP_STATSD_TAGS', '')
...
client.instance_variable_set(:@postfix, ",#{tags}") if tags != ""
```